spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -293,13 +296,7 @@ public void addQueueNames(String... queueNames) {
 	 * @param queues The queue(s) to add.
 	 */
 	public void addQueues(Queue... queues) {
-		Assert.notNull(queues, "'queues' cannot be null");
-		Assert.noNullElements(queues, "'queues' cannot contain null elements");
-		String[] queueNames = new String[queues.length];
-		for (int i = 0; i < queues.length; i++) {
-			queueNames[i] = queues[i].getName();
-		}
-		this.addQueueNames(queueNames);
+		this.addQueueNames(collectQueueNames(queues));
```

Comment:
We can do that on merge.
But since you, @panchenko , are so active contributor, please, consider to follow our code style and overall tracking rules to avoid similar comments again and again.
That should save some time for us, as well as for you 😄